marimo/_plugins/ui/_impl/from_panel.py

```diff
@@ -0,0 +1,255 @@
+
+
+def from_panel(obj: Any) -> UIElement[Any, Any]:
+    """Create a UIElement from an Panel."""
+    try:
+        cached = obj in cache
+        cachable = True
+    except TypeError:
+        cachable = cached = False
+    if cached:
+        ui_element = cache[obj]
+    else:
+        ui_element = panel(obj)
+    if cachable:
+        cache[obj] = ui_element
+    return ui_element
+
+
+T = Dict[str, Any]
+
+
+@dataclass
+class SendToWidgetArgs:
+    message: Any
+    buffers: Optional[Any] = None
+
+
+from pyviz_comms import Comm, CommManager, JupyterCommJS, JupyterComm
+
+
+class MarimoPanelComm(Comm):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._comm = MarimoComm(
+            comm_id=self.id,
+            target_name="panel.comms",
+            data={},
+            comm_manager=COMM_MANAGER,
+        )
+        self._comm.on_msg(self._handle_msg)
+        if self._on_open:
+            self._on_open({})
+
+    @classmethod
+    def decode(cls, msg: SendToWidgetArgs):
+        buffers = {i: v for i, v in enumerate(msg.buffers)}
+        return dict(msg.message, _buffers=buffers)
+
+    def send(self, data=None, metadata=None, buffers=[]):
+        self.comm.send({"content": data}, metadata=metadata, buffers=buffers)
+
+
+class MarimoPanelCommManager(CommManager):
+
+    client_comm = MarimoPanelComm
+    server_comm = MarimoPanelComm
+
+
+def render_extension(load_timeout: int = 500, reloading: bool = False):
+    from bokeh.io.notebook import curstate
+    from panel.config import config
+    from panel.io.notebook import CDN, INLINE, settings, state, _Unset, Resources, bundle_resources, require_components, _autoload_js
+
+    curstate().output_notebook()
+
+    resources = INLINE if config.inline else CDN
+    prev_resources = settings.resources(default="server")
+    user_resources = settings.resources._user_value is not _Unset
+    nb_endpoint = not state._is_pyodide
+    resources = Resources.from_bokeh(resources, notebook=nb_endpoint)
+    try:
+        bundle = bundle_resources(
+            None, resources, notebook=nb_endpoint, reloading=reloading,
+            enable_mathjax='auto'
+        )
+        configs, requirements, exports, skip_imports = require_components()
+        ipywidget = 'ipywidgets_bokeh' in sys.modules
+        bokeh_js = _autoload_js(
+            bundle=bundle,
+            configs=configs,
+            requirements=requirements,
+            exports=exports,
+            skip_imports=skip_imports,
+            ipywidget=ipywidget,
+            reloading=reloading,
+            load_timeout=load_timeout
+        )
+    finally:
+        if user_resources:
+            settings.resources = prev_resources
+        else:
+            settings.resources.unset_value()
+    return bokeh_js
+
+
+@mddoc
+class panel(UIElement[T, T]):
```

Comment:
nice 🎉

Reply:
I guess 😬

Still needs a lot of cleanup I think.